### DISCUSSION TITLE
How to enable disable row along with default checkbox based on data

### DISCUSSION THREAD

**Discussion by hytechpro-bikeshs** [ghiscoding/Angular-Slickgrid#908] [Q&A] [answered]
i am using slickgrid to display paid and unpaid invoices so i want to disable checkbox along with row for paid invoices and enable for unpaid, i didn't find any event where i can overwrite the options
version using 1.9. 6

**Comment by ghiscoding** [comment, marked as answer]
Version 1.x is no longer maintained, and has not been for a very long time (at least 4 years), you should definitely upgrade to the latest version and then you can follow this Wiki [Row Selection](https://github.com/ghiscoding/Angular-Slickgrid/wiki/Row-Selection) to do what you want. I have no clue if the Wiki code works with version 1.x and I will not spend time to find out either. Good Luck

**Reply by hytechpro-bikeshs** [reply]
Actually i using "@angular/core": "~7.0.0",could you please tell which version needs to be installed to enable disable row based on data.
Please help me out.

**Reply by ghiscoding** [reply]
The method that you need is [selectableOverride](https://github.com/ghiscoding/Angular-Slickgrid/wiki/Row-Selection#disable-custom-rows-selections-via-selectableoverride) and that was added in SlickGrid (core) in this [PR](https://github.com/6pac/SlickGrid/commit/72c951a0c02ce9e02c9e41ff1b24a41c3059df1d) (March 2019), so you probably need minimum of Angular-Slickgrid 3.x or you can try to upgrade just SlickGrid itself via your `package.json` but I'm not sure that would work (it might break Angular-Slickgrid) and again I do not support older version of Angular-Slickgrid so you are on your own trying to get this working on old versions. It's definitely a good idea to upgrade, Angular 7.x is EOL (End of Life) support for couple years already, so why stick with it? See Angular [Support policy](https://angular.io/guide/releases#support-policy-and-schedule). If you decide to upgrade, then follow each [Migration Guide](https://github.com/ghiscoding/Angular-Slickgrid/wiki/Migration-to-3.x) of Angular-Slickgrid

**Reply by hytechpro-bikeshs** [reply]
i am able to do enable and disable in same version and working fine in each conditions ,that is alternate solution., Thanks

**Reply by hytechpro-bikeshs** [reply]
@ghiscoding is there any way to handle custom checkbox selection scroll
Actually checkbox get unchecked when scroll down or up.

**Reply by ghiscoding** [reply]
Please ask your questions on Stack Overflow where a lot more people can answer, not here, I don't have time to research and answer everyone's questions.

As per this [SO](https://stackoverflow.com/questions/16357259/slickgrid-checkbox-and-filtering-problems), you need to pass True for the argument `syncGridSelection` of the DataView, you can do this via:
```ts
this.gridOptions = {
  // ... other options
  dataView: { 
    syncGridSelection: true
  }
}
```
but it's already enabled by default so it probably won't do much, unless you're using a Backend Service API (OData, GraphQL), if so then you can try enabling `syncGridSelectionWithBackendService`

https://github.com/ghiscoding/Angular-Slickgrid/blob/965e9ff893f07fdf7323ad374efdbd0ee6a4858f/src/app/modules/angular-slickgrid/global-grid-options.ts#L72-L75

Row selections aren't supposed to get unchecked when scrolling, unless you use Pagination then you might lose when changing page when using Backend Services
